stablehlo/dialect/TypeInference.cpp

```diff
@@ -329,24 +331,28 @@ verifyWindowAttributesAndInferWindowDimensions(
     WindowDimension& dim = window[i];
 
     dim.size = windowDimensions[i];
+    // reduce_window_c5
     if (!isDynamicDimSize(dim.size) && dim.size <= 0)
```

Comment:
You're right, this should be a constant so it should never be dynamic.

Reply:
In that case, let's have a ReduceWindow-specific check that makes sure that its windowDimensions cannot be dynamic?

Reply:
Or not? since attributes can't have dynamic sizes to begin with. Otherwise, we would need to add this check for all ops with attributes, and there isn't a way to encode `?` in an attribute.

Reply:
"there isn't a way to encode ? in an attribute to test it". It is possible to use `-9223372036854775808` as a value of the window_dimensions attribute. For example, this passes verification right now:

```mlir
func.func @main(%arg0: tensor<4x6xi16>, %arg1: tensor<i16>) -> tensor<3x5xi16> {
  %0 = "stablehlo.reduce_window"(%arg0, %arg1) ({
  ^bb0(%arg2: tensor<i16>, %arg3: tensor<i16>):
    %5 = stablehlo.add %arg2, %arg3 : tensor<i16>
    stablehlo.return %5 : tensor<i16>
  }) {
    window_dimensions = dense<-9223372036854775808> : tensor<2xi64>
  } : (tensor<4x6xi16>, tensor<i16>) -> tensor<3x5xi16>
  func.return %0 : tensor<3x5xi16>
}
```

Reply:
That's what I wanted to avoid. IIRC we were using -1 as `kDynamic` previously, so I wanted to avoid using that number.   But upon further look, it looks like we do have instances of using `-9223372036854775808` elsewhere. I guess in the unlikely chance this value changes, it will be handled in the future.

Reply:
Since we don't have this check for all ops just yet, I've added #1472 to address this as part of spec compliance. I'll add this check for this op for now.

Reply:
If the fix is intended to focus on just ReduceWindowOp, then I think it's premature to change `convert1DAttribute` and `convertPaddingAttribute` because they are used in other ops as well. How about we revert the changes to these functions in this PR and leave everything to #1472?